Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/MarkerDetection/MarkerVisualizeCoordinateService.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.MarkerDetection;
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.Sharing;
+using System;
+using System.Numerics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    /// <summary>
+    /// A variant of marker based <see cref="ISpatialCoordinateService"/> implementation. This one tracks coordinates displayed on the screen of current mobile device.
+    /// The logic is that every time you start tracking a new coordinate is created and shown on the screen, after you stop tracking that coordinates location is no longer updated with the device.
+    /// </summary>
+    public class MarkerVisualizeCoordinateService : SpatialCoordinateServiceBase<int>
+    {
+        private class SpatialCoordinate : SpatialCoordinateBase<int>
+        {
+            private UnityEngine.Matrix4x4 worldToCoordinate;
+            private UnityEngine.Quaternion worldToCoordinateRotation;
+            private UnityEngine.Quaternion coordinateToWorldRotation;
+
+            public new bool IsLocated
+            {
+                get => base.IsLocated;
+                set => base.IsLocated = value;
+            }
+
+            public UnityEngine.Matrix4x4 WorldToCoordinate
+            {
+                set
+                {
+                    worldToCoordinate = value;
+                    worldToCoordinateRotation = UnityEngine.Quaternion.LookRotation(value.GetColumn(2), value.GetColumn(1));
+                    coordinateToWorldRotation = UnityEngine.Quaternion.Inverse(worldToCoordinateRotation);
+                }
+            }
+
+            public SpatialCoordinate(int id)
+                : base(id) { }
+
+            public override Vector3 CoordinateToWorldSpace(Vector3 vector) => worldToCoordinate.inverse.MultiplyPoint(vector.AsUnityVector()).AsNumericsVector();
+
+            public override Quaternion CoordinateToWorldSpace(Quaternion quaternion) => (coordinateToWorldRotation * quaternion.AsUnityQuaternion()).AsNumericsQuaternion();
+
+            public override Vector3 WorldToCoordinateSpace(Vector3 vector) => worldToCoordinate.MultiplyPoint(vector.AsUnityVector()).AsNumericsVector();
+
+            public override Quaternion WorldToCoordinateSpace(Quaternion quaternion) => (worldToCoordinateRotation * quaternion.AsUnityQuaternion()).AsNumericsQuaternion();
+        }
+
+        private readonly IMarkerVisual markerVisual;
+        private readonly UnityEngine.Transform markerInWorldSpace;
+        private readonly Func<int> generateMarkerId;
+
+        private SpatialCoordinate markerBeingVisualized;
```

Comment:
>markerBeingVisualized [](start = 34, length = 21)

nit: maybe just markerCoordinate?

Reply:
Yeah, will update as well.

---
In reply to: [270509711](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3732#discussion_r270509711) [](ancestors = 270509711)